app/Main.hs

```diff
@@ -1,6 +1,85 @@
 module Main where
 
-import Text.Printf (printf)
+import Entities
+import Control.Monad.IO.Class  (liftIO)
+import Database.Persist
+import Database.Persist.Sqlite
+import Text.Megaparsec
+import FooParser (parseSections, Section(..))
+import qualified Database.Esqueleto as E
 
 main :: IO ()
-main = printf "foo"
+main = do
+  fileContents <- readFile "specs.foo"
+
+  -- parse specs.foo
+  let sections = case runParser parseSections "specs.foo" fileContents of
+        Left err        -> error $ "could not parse .foo file with error: " ++ show err
+        Right sections' -> sections'
+
+  -- run everything inside single connection/txn against an in-memory
+  -- instance of SQLite
+  runSqlite ":memory:" $ do
+
+    -- run a safe migration
+    runMigration migrateAll
+
+    -- prepare users, insert users, get entities
+    let users = map (\section -> User $ sectionUsername section) sections :: [User]
```

Comment:
The target audience here isn't supposed to be total novices, right? Maybe bring up eta reduction and point free style here because they'll definitely end up seeing that in the wild.